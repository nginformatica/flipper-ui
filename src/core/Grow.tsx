import { Grow as MuiGrow } from '@material-ui/core'
import React from 'react'
import { DefaultProps } from './types'

interface GrowProps extends DefaultProps {
    in: boolean
    timeout?: number | { enter?: number; exit?: number } | 'auto'
    children?: React.ReactElement<Record<string, unknown>>
}

const Grow = ({
    children,
    padding,
    margin,
    style = {},
    ...otherProps
}: GrowProps) => (
    <MuiGrow style={{ padding, margin, ...style }} {...otherProps}>
        {children}
    </MuiGrow>
)

export default Grow
