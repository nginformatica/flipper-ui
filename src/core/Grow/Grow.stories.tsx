import React, { useState } from 'react'
import { Meta } from '@storybook/react'
import Grow from '.'
import Button from '../Button'

export default {
    title: 'Grow',
    component: Grow,
    argTypes: {
        backgroundColor: { control: 'color' }
    }
} as Meta<typeof Grow>

export const Default = () => {
    const [open, setOpen] = useState(true)
    const [label, setLabel] = useState('Close')

    function handleClick() {
        setOpen(!open)
        label === 'Close' ? setLabel('Open') : setLabel('Close')
    }

    return (
        <>
            <Button
                color='primary'
                variant='contained'
                onClick={handleClick}
                style={{ marginRight: '3em' }}>
                {label}
            </Button>
            <Grow in={open} timeout={{ enter: 500, exit: 500 }}>
                <span>I am open for discussions</span>
            </Grow>
        </>
    )
}
