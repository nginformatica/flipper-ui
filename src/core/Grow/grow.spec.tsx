import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Grow from '.'

describe('Fade', () => {
    it('should render closed', () => {
        render(
            <Grow in={false}>
                <span>Grow Text</span>
            </Grow>
        )
        const element = screen.getByText('Grow Text')

        expect(element).toHaveProperty('style.visibility', 'hidden')
        expect(element).toHaveProperty('style.opacity', '0')
    })

    it('should render opened', () => {
        render(
            <Grow in>
                <span>Grow Text</span>
            </Grow>
        )

        const element = screen.getByText('Grow Text')

        expect(element).not.toHaveProperty('style.visibility', 'hidden')
        expect(element).toHaveProperty('style.opacity', '1')
    })

    it('should match snapshot', () => {
        const { container: opened } = render(
            <Grow in>
                <span>Grow Text</span>
            </Grow>
        )

        const { container: closed } = render(
            <Grow in={false}>
                <span>Grow Text</span>
            </Grow>
        )

        expect(opened).toMatchSnapshot()
        expect(closed).toMatchSnapshot()
    })
})
