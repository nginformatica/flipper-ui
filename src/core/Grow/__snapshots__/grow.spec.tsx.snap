// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fade should match snapshot 1`] = `
<div>
  <span
    style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    Grow Text
  </span>
</div>
`;

exports[`Fade should match snapshot 2`] = `
<div>
  <span
    style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
  >
    Grow Text
  </span>
</div>
`;
